### DISCUSSION TITLE
Login to app using maestro

### DISCUSSION THREAD

**Discussion by vikaskrsharma** [mobile-dev-inc/Maestro#1011] [Q&A] [answered]
I have a app where login is mandatory using mobile phone and otp. for testing apps, our otp saved in redis database and i am fetching it on runtime using java library. How can this be acheived in maestro.

**Comment by axelniklasson** [comment, marked as answer]
Hi @vikaskrsharma! You can use Javascript to fetch the OTP and use it in your tests. You can refer to [this example](https://maestro.mobile.dev/examples/advanced-wikipedia-android#authentication) for some inspiration. Thanks!
